src/distinguished_name.cpp

```diff
@@ -84,22 +92,52 @@ DistinguishedName DistinguishedName::fromX509Name(X509_NAME *x509)
     return builder.build();
 }
 
-auto _createTuple(const DistinguishedName &dn)
+auto _createTuple(const DistinguishedName& dn)
 {
-    return std::tie(dn.commonName(),
+    return std::make_tuple(dn.commonName(),
                     dn.countryName(),
-                    dn.localityName(),
-                    dn.stateOrProvinceName(),
-                    dn.organizationalUnitName(),
-                    dn.organizationName(),
-                    dn.pkcs9EmailAddress(),
-                    dn.serialNumber(),
-                    dn.userId());
+                     dn.localityName(),
+                     dn.stateOrProvinceName(),
+                     dn.organizationalUnitName(),
+                     dn.organizationName(),
+                     dn.pkcs9EmailAddress(),
+                     dn.serialNumber(),
+                     dn.userId(),
+                     dn.givenName());
 }
 
 bool DistinguishedName::operator==(const DistinguishedName &other) const
 {
-    return _createTuple(*this) == _createTuple(other);
+    if (_customAttributeOrderFlag) {
+        // order matters now
+        return std::equal(_attributes.begin(), _attributes.end(), other._attributes.begin(), other._attributes.end()
```

Comment:
Doesn't the struct have an auto-generated equality operator that does the same thing? Also, doesn't `std::vector` have an equality operator, which would make this entire block equal to `_attributes == other._attributes`?